### PR TITLE
Update MoveToMapPokemon to use events instead of logger (similar to #2867)

### PR DESCRIPTION
**Pull request by chrisle**
Short Description: 

Update `MoveToMapPokemon` to use events instead of logger. Same as [PR 2867](https://github.com/PokemonGoF/PokemonGo-Bot/pull/2867) but includes documentation and a refactor of the module.

Fixes:
- Deprecation warnings emitted by MoveToMapPokemon.

### CODE REVIEW

**Comment by bixuanzju** [comment]
This is better than mine 👍

**Comment by bigkraig** [comment]
maybe a squash and a rebase is in order, i can't apply this cleanly

**Comment by chrisle** [comment]
I was just realizing that actually. Hold on.

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -383,6 +383,35 @@ def _register_events(self):
             parameters=('nickname',)
         )
         self.event_manager.register_event('unset_pokemon_nickname')
+        
+        # Move To map pokemon
+        self.event_manager.register_event(
+            'move_to_map_pokemon_fail',
+            parameters=('message')
```
Comment (douglascamata) [inline pokemongo_bot/__init__.py]: this needs to be `('message',)`, otherwise the tuple isn't valid

**Comment by douglascamata** [comment]
👍 

@TheSavior review this when you can, please.

**Comment by brantje** [comment]
Can you resolve the conflicts?

**Comment by chrisle** [comment]
*grr.* Some help needed: How do I squish or rebase commits that aren't mine so I'm left with just my commits?

**Comment by douglascamata** [comment]
@chrisle easier to clone the project again, copy over your files and open another PR, imho.

**Comment by chrisle** [comment]
Yeah.. i was starting to think that. OK.. hold on.

**Comment by chrisle** [comment]
Please disregard this pull request. The merge went south. See [PR 2913](https://github.com/PokemonGoF/PokemonGo-Bot/pull/2913) instead for cleaner pull request.

**Comment by bigkraig** [comment]
You can rebuild your local branch and force push it to github instead of opening a new pr

Also used rebase -i to squash the commits

**Pull request closed without merging**